examples/ui/drag_drop.rs

```diff
@@ -0,0 +1,374 @@
+use bevy::{
+    input::mouse::{MouseButtonInput, MouseMotion},
+    prelude::*,
+    render::texture::ImageSettings,
+    ui::FocusPolicy,
+};
+
+// Markers
+#[derive(Component)]
+struct Dragable;
+
+#[derive(Component)]
+struct DragActive;
+
+#[derive(Component)]
+struct DragMoving;
+
+#[derive(Component)]
+struct DragDropping;
+
+#[derive(Component)]
+struct DragStarting;
+
+#[derive(Component)]
+struct SourceDropContainer(pub Entity);
+
+#[derive(Component)]
+pub struct DropContainer(bool);
+
+impl Default for DropContainer {
+    fn default() -> Self {
+        DropContainer(true)
+    }
+}
+
+struct DragableHoverEvent(pub Entity);
+struct DragableClickEvent(pub Entity);
+struct DragableDropEvent(pub Entity);
+
+#[derive(Component)]
+struct MouseIn;
+
+struct MouseEnterEvent(pub Entity);
+struct MouseExitEvent(pub Entity);
+
+struct DragDropPlugin;
+
+pub const CLEAR_COLOR: Color = Color::rgb(0.1, 0.1, 0.1);
+
+impl Plugin for DragDropPlugin {
+    fn build(&self, app: &mut App) {
+        app.insert_resource(ImageSettings::default_nearest())
+            .insert_resource(ClearColor(CLEAR_COLOR))
+            .add_plugins(DefaultPlugins)
+            .add_startup_system_to_stage(StartupStage::PreStartup, spawn_camera)
+            .add_startup_system_to_stage(StartupStage::PostStartup, spawn_scene)
+            .add_event::<DragableClickEvent>()
+            .add_event::<DragableHoverEvent>()
+            .add_event::<DragableDropEvent>()
+            .add_event::<MouseEnterEvent>()
+            .add_event::<MouseExitEvent>()
+            // System order is important
```

Comment:
System insertion order doesn't act how you might expect; only explicit ordering works.

I'd probably remove this comment completely.

Reply:
I was refering to the .after statements and not the actual order of insertion but seems like the comment distracted instead of adding information :D